packages/sync-actions/src/shipping-methods-actions.js

```diff
@@ -0,0 +1,91 @@
+import forEach from 'lodash.foreach'
+import { buildBaseAttributesActions } from './utils/common-actions'
+
+export const baseActionsList = [
+  { action: 'setKey', key: 'key' },
+  { action: 'changeName', key: 'name' },
+  { action: 'setDescription', key: 'description' },
+  { action: 'changeIsDefault', key: 'isDefault' },
+  { action: 'setPredicate', key: 'predicate' },
+  { action: 'changeTaxCategory', key: 'taxCategory' },
+]
+
+export function actionsMapBase(diff, oldObj, newObj) {
+  return buildBaseAttributesActions({
+    actions: baseActionsList,
+    diff,
+    oldObj,
+    newObj,
+  })
+}
+
+export function actionsMapZoneRates(diff, oldObj, newObj) {
+  const actions = []
+  if (!diff.zoneRates) return actions
+
+  const zoneActions = []
+  const shippingRateActions = []
+
+  forEach(diff.zoneRates, (zoneRate, index) => {
+    // updating zones
+    if (Array.isArray(zoneRate)) {
+      if (zoneRate.length === 3) {
+        if (zoneRate[2] !== 3) {
+          zoneActions.push({
+            action: 'removeZone',
+            zone: zoneRate[0].zone,
+          })
+        }
+      } else if (zoneRate.length === 1) {
+        zoneActions.push({
+          action: 'addZone',
+          zone: zoneRate[0].zone,
+        })
+      }
+    } else if (zoneRate.zone) {
+      if (Array.isArray(zoneRate.zone.id)) {
+        zoneActions.push({
+          action: 'removeZone',
+          zone: oldObj.zoneRates[index].zone,
+        })
+        zoneActions.push({
+          action: 'addZone',
+          zone: newObj.zoneRates[index].zone,
+        })
+      }
+      // updating shippingRates
+    } else if (zoneRate.shippingRates) {
+      forEach(zoneRate.shippingRates, (shippingRate, subIndex) => {
```

Comment:
The nested loops feel a bit off but I fear we'll introduce more complexity when removing them than having when keeping 'em. It's due mostly to the fact that you seem to need access to the `oldObj.zoneRates[index].zone` which you would likely not have if you'd flatten/pick all `shippingRates` from all `zones` unless you add the "backward" index to each picked `shippingRates`. In that scenario you'll not reduce the iterations (as in Big O) after all I think. /cc @wizzy25